### PR TITLE
#19189 : A list with an empty String representing an empty value in t…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…he Tag field is being sent from the Edit Mode in Angular. This is causing problems in our current Java code, as passing down an empty String to the `CategoryAPI.find` method returns a null value, which causes a Null Pointer Exception during the checkin process.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java [hunk 1]
```diff
@@ -423,11 +423,12 @@ private void saveWebAsset(Map<String, Object> contentletFormData,
 			}
 		 }
 
-		if (categoriesList != null && categoriesList.size() > 0) {
-			for (final Iterator iterator = categoriesList.iterator(); iterator.hasNext();) {
-
-				final String tmpString = (String) iterator.next();
-				categories.add(catAPI.find(tmpString, user, false));
+		if (UtilMethods.isSet(categoriesList)) {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java:426]: Should the condition be:

```
if (UtilMethods.isSet(categoriesList) && categoriesList.size() > 0){
```

Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java:426]: @wezell the `isSet` method already checks for both null and empty list scenarios:
```java
public static final boolean isSet(final Collection<?> collection) {
    return null != collection && !collection.isEmpty();
}
```

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java [hunk 2]
```diff
@@ -423,11 +423,12 @@ private void saveWebAsset(Map<String, Object> contentletFormData,
 			}
 		 }
 
-		if (categoriesList != null && categoriesList.size() > 0) {
-			for (final Iterator iterator = categoriesList.iterator(); iterator.hasNext();) {
-
-				final String tmpString = (String) iterator.next();
-				categories.add(catAPI.find(tmpString, user, false));
+		if (UtilMethods.isSet(categoriesList)) {
+			for (final Iterator<String> iterator = categoriesList.iterator(); iterator.hasNext();) {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java:427]: I would replace the iterator with a normal `for` over `categoriesList` 
```for (final String categoryId : categoriesList)```

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/web/ContentletWebAPIImpl.java:427]: Yes, I thought about it too. But didn't add many changes as this is very old code that will be moved over to a REST Endpoint, hopefully soon.

**Review by erickgonzalez — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**